extensions/amp-a4a/0.1/real-time-config-manager.js

```diff
@@ -66,327 +66,364 @@ export const RTC_ERROR_ENUM = {
-        user().warn(TAG, `Unknown macro: ${macro} for vendor: ${vendor}`);
+        dev().warn(TAG, `Invalid url: ${urlObj}`);
       }
+      this.inflateAndSendRtc_(url,
+          customMacros,
+          errorReportingUrl);
     });
-    // The ad network defined macros override vendor defined/pub specifed.
-    const macros = Object.assign(validVendorMacros, customMacros);
-    inflateAndSendRtc_(a4aElement, url, seenUrls, promiseArray, rtcStartTime,
-        macros, rtcConfig['timeoutMillis'], errorReportingUrl,
-        vendor.toLowerCase());
-  });
-  return Promise.all(promiseArray);
-}
+  }
 
-/**
- * @param {!./amp-a4a.AmpA4A} a4aElement
- * @param {string} url
- * @param {!Object<string, boolean>} seenUrls
- * @param {!Array<!Promise<!rtcResponseDef>>} promiseArray
- * @param {number} rtcStartTime
- * @param {!Object<string, !../../../src/service/variable-source.AsyncResolverDef>} macros
- * @param {number} timeoutMillis
- * @param {string} errorReportingUrl
- * @param {string=} opt_vendor
- * @private
- */
-export function inflateAndSendRtc_(a4aElement, url, seenUrls, promiseArray,
-  rtcStartTime, macros, timeoutMillis, errorReportingUrl, opt_vendor) {
-  const {win} = a4aElement;
-  const ampDoc = a4aElement.getAmpDoc();
-  const callout = opt_vendor || getCalloutParam_(url);
-  const checkStillCurrent = a4aElement.verifyStillCurrent.bind(a4aElement)();
   /**
-   * The time that it takes to substitute the macros into the URL can vary
-   * depending on what the url requires to be substituted, i.e. a long
-   * async call. Thus, however long the URL replacement took is treated as a
-   * time penalty.
+   * Manages sending the RTC callouts for all specified vendors.
+   * @param {!Object<string, !../../../src/service/variable-source.AsyncResolverDef>} customMacros The ad-network specified macro
    */
-  const send = url => {
-    if (Object.keys(seenUrls).length == MAX_RTC_CALLOUTS) {
-      return buildErrorResponse_(
-          RTC_ERROR_ENUM.MAX_CALLOUTS_EXCEEDED,
-          callout, errorReportingUrl, win, ampDoc);
-    }
-    if (!isSecureUrl(url)) {
-      return buildErrorResponse_(RTC_ERROR_ENUM.INSECURE_URL,
-          callout, errorReportingUrl, win, ampDoc);
-    }
-    if (seenUrls[url]) {
-      return buildErrorResponse_(RTC_ERROR_ENUM.DUPLICATE_URL,
-          callout, errorReportingUrl, win, ampDoc);
-    }
-    seenUrls[url] = true;
-    if (url.length > MAX_URL_LENGTH) {
-      url = truncUrl_(url);
-    }
-    return sendRtcCallout_(
-        url, rtcStartTime, win, timeoutMillis, callout, checkStillCurrent,
-        errorReportingUrl, ampDoc);
-  };
+  handleRtcForVendorUrls(customMacros) {
+    // For each vendor the publisher has specified, inflate the vendor
+    // url if it exists, and send the RTC request.
+    Object.keys(this.rtcConfig_['vendors'] || []).forEach(vendor => {
+      const vendorObject = RTC_VENDORS[vendor.toLowerCase()];
+      const url = vendorObject ? vendorObject.url : '';
```

Comment:
In one case you access via . notation (vendorObject.url) and other via quote (vendorObject['errorReportingUrl']), please be consistent (dot notation should work given this has been externed, right?)